pkg/authorization/authz_unix_test.go

```diff
@@ -135,6 +137,48 @@ func TestResponseModifier(t *testing.T) {
 	}
 }
 
+func TestDrainBody(t *testing.T) {
+
+	// Small message size
+	body, closer, err := drainBody(ioutil.NopCloser(bytes.NewReader([]byte(strings.Repeat("a", 10)))))
+
+	if len(body) != 10 {
+		t.Fatalf("Body must be copied, actual length '%d'", len(body))
+	}
+	if closer == nil {
+		t.Fatalf("Closer must not be nil")
+	}
+	if err != nil {
+		t.Fatalf("Error must not be nil '%v'", err)
+	}
+
+	// Max size message
+	body, closer, err = drainBody(ioutil.NopCloser(bytes.NewReader([]byte(strings.Repeat("a", maxBodySize-1)))))
+
+	if len(body) != maxBodySize-1 {
+		t.Fatalf("Body must be copied, actual length '%d'", len(body))
+	}
+	if closer == nil {
+		t.Fatalf("Closer must not be nil")
+	}
+	if err != nil {
+		t.Fatalf("Error must not be nil '%v'", err)
+	}
+
+	// Large request
+	body, closer, err = drainBody(ioutil.NopCloser(bytes.NewReader([]byte(strings.Repeat("a", maxBodySize*2)))))
+
+	if len(body) != 0 {
+		t.Fatalf("Body must not be copied (request is too lrage), actual length '%d'", len(body))
+	}
+	if closer == nil {
```

Comment:
Should this also check that `closer` still points to the full data provided to `drainBody()` and not truncated at `maxBodySize`?